components/shell/ConfigurationFlow.tsx

```diff
@@ -1,59 +1,112 @@
 'use client'
-import { ComponentProps, ReactElement, cloneElement, useState } from 'react'
-import Link from 'next/link'
+
 import {
   ArrowLeftOnRectangleIcon,
-  ArrowRightOnRectangleIcon,
   ArrowRightCircleIcon,
+  ArrowRightOnRectangleIcon,
 } from '@heroicons/react/20/solid'
 import { useMutation } from '@tanstack/react-query'
-import { Loading } from '@/common/Loader'
-import client, { ClientSession } from '@/lib/client'
+import Link from 'next/link'
+import { ComponentProps, ReactElement, cloneElement, useState } from 'react'
+
+import { ErrorInfo } from '@/common/ErrorInfo'
 import { Checkbox, Input } from '@/common/forms'
+import { Loading } from '@/common/Loader'
 import {
+  OzoneConfig,
   OzoneConfigFull,
   getServiceUrlFromDoc,
   withDocAndMeta,
 } from '@/lib/client-config'
-import { ErrorInfo } from '@/common/ErrorInfo'
-import { useSession } from '@/lib/useSession'
+import { Agent } from '@atproto/api'
+import { useAuthContext, useAuthDid, useAuthIdentifier } from './AuthContext'
+import { ConfigurationState, ReconfigureOptions } from './ConfigurationContext'
 
-export function ConfigurationFlow() {
-  const session = useSession()
+type ReconfigureFn = (options?: ReconfigureOptions) => void | Promise<void>
 
-  if (!session) {
+export type ConfigurationFlowProps = {
+  state: ConfigurationState
+  config: OzoneConfig | undefined
+  error: Error | null
+  labelerAgent: Agent | undefined
+  reconfigure: ReconfigureFn
+}
+
+export function ConfigurationFlow({
+  state,
+  config,
+  error,
+  labelerAgent,
+  reconfigure,
+}: ConfigurationFlowProps) {
+  const { signOut } = useAuthContext()
+
+  const authDid = useAuthDid()
+  const authIdentifier = useAuthIdentifier()
+
+  if (state === ConfigurationState.Ready) {
+    return <Loading message="Redirecting..." />
+  }
+
+  if (state === ConfigurationState.Unauthorized) {
     return (
       <>
         <ErrorInfo type="warn" className="mt-2">
-          {`You're`} not logged-in. Please login using your Ozone service
-          account in order to configure Ozone.
+          Your credentials are not authorized on this network. Please login as
+          an authorized user in order to configure Ozone.
```

Comment:
We specifically want the user to login using their service account, i.e. the account with the same DID as their labeler.  I don't think we need to iterate on the verbiage of the configuration flow as part of the OAuth integration, unless there's some new auth state that needs to be incorporated into the configuration flow.

Reply:
```suggestion
          {`You're`} not logged-in. Please login using your Ozone service
          account in order to configure Ozone.
```